### PR TITLE
Feat/add notification route to gateway

### PR DESCRIPTION
**Pull request by Nhsdkk**
1) Add GET: /notification/{notification_id} route to get notification by id
2) Add GET: /notification/admin/all/ route to get all notifications
3) Add GET: /notification/my/ route to get user's notifications
4) Add POST: /notification/ route to create notification
5) Add PUT: /notification/ route to update notification data as author
6) Add PUT: /notification/admin/ route to update notification as admin
7) Add DELETE: /notification/ route to delete notification

### CODE REVIEW

File: gateway/app/grpc_client/grpc_client.py [hunk 1]
```diff
@@ -41,13 +41,13 @@ def __init__(self, host: str, port: int, stub: type[T]) -> None:
...
-        """
+        # """
+        # Get request client
+        #
+        # Returns
+        # -------
+        # T
+        #     Request client
+        #
+        # """
```
Comment (MaxPlays35) [inline gateway/app/grpc_client/grpc_client.py]: It looks weird

File: gateway/app/models/notification.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        Get notification instance from proto object
+    to_proto()
+        Get proto object from Notification instance
+
+    """
+
+    id: Annotated[str, Field("", min_length=1), AfterValidator(str_special_characters_validator)]
+    event_id: Annotated[str, Field("", min_length=1), AfterValidator(str_special_characters_validator)]
+    author_id: Annotated[str, Field("", min_length=1), AfterValidator(str_special_characters_validator)]
+    enabled: bool
```
Comment (ZixMai) [inline gateway/app/models/notification.py]: must be with default value = True

File: gateway/app/models/notification.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+        """
+        notification = GrpcNotification(
+            id=self.id,
+            event_id=self.event_id,
+            author_id=self.author_id,
+            enabled=self.enabled,
+        )
+
+        notification.created_at.FromDatetime(self.created_at)
```
Comment (ZixMai) [inline gateway/app/models/notification.py]: needs to be with utc tz

File: gateway/app/models/notification.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        if self.deleted_at is not None:
+            notification.deleted_at.FromDatetime(self.deleted_at)
```
Comment (ZixMai) [inline gateway/app/models/notification.py]: needs to be with utc tz

**Review by ZixMai — CHANGES_REQUESTED** [review]

**Review by ZixMai — APPROVED** [review]

**Review by ZixMai — APPROVED** [review]

**Review by MaxPlays35 — APPROVED** [review]

**Review by ZixMai — APPROVED** [review]

**Pull request merged**